master/buildbot/util/git.py

```diff
@@ -104,7 +104,23 @@ def getSshWrapperScriptContents(keyPath, knownHostsPath=None):
     return '#!/bin/sh\n{0} "$@"\n'.format(ssh_command)
 
 
+def isBareSshHostKey(key):
+    # See RFC 4253
+    formats = [
```

Comment:
What about ed25519 or ecdsa keys?

Reply:
Looks like the number of formats is so large that it might be hard to distinguish between known_hosts file and a bare key. I've re-implemented the feature as a separate parameter due to this reason.